libtrap/src/ifc_tcpip.c

```diff
@@ -1637,77 +1679,93 @@ int create_tcpip_sender_ifc(trap_ctx_priv_t *ctx, const char *params, trap_outpu
       goto failsafe_cleanup;
    }
 
-   priv->ctx = ctx;
-   priv->socket_type = type;
-   priv->ifc_idx = idx;
-
    /* Parsing params */
    param_iterator = trap_get_param_by_delimiter(params, &server_port, TRAP_IFC_PARAM_DELIMITER);
    if ((server_port == NULL) || (strlen(server_port) == 0)) {
       VERBOSE(CL_ERROR, "Missing 'port' for %s IFC.", (type == TRAP_IFC_TCPIP ? "TCPIP" : "UNIX socket"));
       result = TRAP_E_BADPARAMS;
       goto failsafe_cleanup;
    }
-   if (param_iterator != NULL) {
-      /* still having something to parse... */
-      param_iterator = trap_get_param_by_delimiter(param_iterator, &max_clients, TRAP_IFC_PARAM_DELIMITER
-);
-   }
-   if (max_clients == NULL) {
-      /* 2nd parameter became optional, set default value when missing */
-      max_num_client = TRAP_IFC_DEFAULT_MAX_CLIENTS;
-   } else {
-      if (sscanf(max_clients, "%u", &max_num_client) != 1) {
-         VERBOSE(CL_ERROR, "Optional max client number given, but it is probably in wrong format.");
-         max_num_client = TRAP_IFC_DEFAULT_MAX_CLIENTS;
+
+   /* Optional params */
+   while (param_iterator != NULL) {
+      param_iterator = trap_get_param_by_delimiter(param_iterator, &param_str, TRAP_IFC_PARAM_DELIMITER);
+      if (param_str == NULL)
+         continue;
+      if (strncmp(param_str, "buffer_count=x", 13) == 0) {
+         if (sscanf(param_str+13, "%u", &buffer_count) != 1) {
```

Comment:
These magic constants like `param_str + 13` are bad practice. Use defined constants, so the reader is aware what the code is trying to do.